snappy/configUpdate.py

```diff
@@ -1,9 +1,21 @@
 # Updates a configuration with new values
 def configUpdate(original, update):
   for key in update:
-    if (key in original and
-        isinstance(original[key], dict) and
-        isinstance(update[key], dict)):
+    if key not in original:
+      raise KeyError("Attempt to update key {} which does not exist".format(key))
+    if isinstance(original[key], dict):
+      if not isinstance(update[key], dict):
+        raise TypeError("Attempt to update key {} with a non dict type".format(key))
       configUpdate(original[key], update[key])
-    else:
+    elif isinstance(original[key], list):
+      if not isinstance(update[key], list):
+        raise TypeError("Attempt to update key {} with a non list type".format(key))
+      original[key] = update[key]
+    elif isinstance(original[key], int):
+      original[key] = int(update[key])
+    elif isinstance(original[key], float):
+      original[key] = float(update[key])
+    elif isinstance(original[key], basestring):
+      if not isinstance(update[key], basestring):
+        raise TypeError("Attempt to update key {} with a non string type".format(key))
       original[key] = update[key]
```

Comment:
This should be an easy function to add tests for--especially the edge cases. Definitely worth doing.